### DISCUSSION TITLE
minimal example to show myst parser work for linking one md file in project A to link to another md file in project B

### DISCUSSION THREAD

**Discussion by monajalal** [executablebooks/MyST-Parser#958] [General]
Could you please provide a mininal example where I can create a section (header with ## as an example) in project A and when I write it in project B, it would be linkable?

Currently, mine is not linkable.

Project A, p_A.md

```
[dino link](#dino-header)
```

Project B, p_B.md

```
## Dino Header {#dino-header}
```

But current no linking happens.

Please note I am using breathe, exhale, and intersphinx for linking project A and B (one depends on the other but shouldn't matter much in this example) in conf.py and index.rst

**Comment by agoose77** [comment]
The reference target syntax you're using here (`{#dino-header}`) is one of two possible target-definition syntaxes in `myst-parser`. This form, using attributes, is opt-in and requires you to enable the `attrs_block` and/or `attrs_inline` extensions. In this case, enabling those extensions won't fix the problem; you need to use a block-level attribute for headings, e.g.
```markdown
{#dino-header}
## Dino Header
```

Or
```markdown
(dino-header)=
## Dino Header
```

**Comment by monajalal** [comment]
@agoose77 Thanks for your suggestion. I added these in my conf.py

```
# https://myst-parser.readthedocs.io/en/latest/syntax/optional.html
myst_enable_extensions = [
    "amsmath",
    "attrs_inline",
    "colon_fence",
    "deflist",
    "dollarmath",
    "fieldlist",
    "html_admonition",
    "html_image",
    "linkify",
    "replacements",
    "smartquotes",
    "strikethrough",
    "substitution",
    "tasklist",
]

myst_heading_anchors = 3
```

and I have:
```
[dino link](#dino-header)

```

in page B.

and 

```

## Dino Header {#dino-header}

```
in page A.

Still page B doesn't link to page A.


Could you please guide why the linking is not working?

**Reply by chrisjsewell** [reply]
Heya, for attributes, you need to add `attrs_block`, and it had to be added above the block that you want it to apply to: https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#block-attributes

But... from your question, I don't think you actually need to use attributed:
Project A, automatically creates an intersphinx record for each document, so then assuming you have intersphinx setup in project B to link to project A (with key `projecta`), you would link to `mydoc.md` (or indeed `mydoc.rst`):

```
[dino link](inv:projecta:std:doc#mydoc)
```

**Reply by chrisjsewell** [reply]
or use the https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#role-external:

```
{external+projecta:std:doc}`dino link <mydoc>`
```

**Reply by chrisjsewell** [reply]
this is documented in https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#cross-project-intersphinx-links

**Comment by michaelweinold** [comment]
Related:

I am using `intersphinx` to link to sections and parts of the API documentation of my main page. That works fine.

But section headings defined like so:

```
(installing-pip)=
## Installing Brightway using `pip`
```

are not picked up by the cross-referencing syntax:

```
<inv:bw#installing-pip>
```

Instead throwing an error

```
WARNING: No matches for 'bw:*:*:installing-pip'
```

Any ideas how to use intersphinx to link to specific sections, @chrisjsewell?

**Reply by chrisjsewell** [reply]
Have you tried using `myst-inv` to look at the source? https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#cross-project-intersphinx-links

I tried this with `myst-inv https://myst-parser.readthedocs.io/en/latest` and it all works fine

**Reply by michaelweinold** [reply]
I did check:

```bash
$ myst-inv https://docs.brightway.dev/en/latest -n index 
name: Brightway
version: latest
base_url: https://docs.brightway.dev/en/latest
objects:
  std:
    doc:
      index:
        loc: index.html
        text: Brightway LCA Software Framework
```

But whether I include the optional `key`, `domain` and `type` or not - section headings which are defined using the `()=` syntax cannot be found.

**Reply by chrisjsewell** [reply]
If you run `myst-inv https://docs.brightway.dev/en/latest -o label`, you can see that `installing-pip` is there in the inventory

**Reply by michaelweinold** [reply]
...ok - maybe it just took a little while for the newly added section reference to be picked up? It works for me as well now. Thank you!
